packages/@cdktf/cli-core/src/lib/dependencies/package-manager.ts

```diff
@@ -629,6 +634,88 @@ class MavenPackageManager extends PackageManager {
   }
 }
 
+class GradlePackageManager extends JavaPackageManager {
+  public static isGradleProject(workingDirectory: string): boolean {
+    const buildGradlePath = path.join(workingDirectory, "build.gradle");
+
+    try {
+      fs.accessSync(buildGradlePath, fs.constants.R_OK | fs.constants.W_OK);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  public async addPackage(
+    packageName: string,
+    packageVersion = "latest.release"
+  ): Promise<void> {
+    const dependencyRegex =
+      /dependencies {(\s*(implementation|api) ('|")(\S)*('|")\s*\n)*}/gm;
+
+    const buildGradlePath = path.join(this.workingDirectory, "build.gradle");
+    const buildGradle = await fs.readFile(buildGradlePath, "utf8");
+
+    const dependencies = dependencyRegex.exec(buildGradle);
+    if (!dependencies || !dependencies.index) {
+      throw Errors.Usage(
+        "Could not find dependencies section in the build.gradle"
+      );
+    }
+
+    // Find the index of the opening bracket of the dependencies section
+    const dependenciesStartIndex = buildGradle.indexOf("{", dependencies.index);
+    const newBuildGradle =
+      buildGradle.slice(0, dependenciesStartIndex + 1) +
+      `\n    implementation '${packageName}:${packageVersion}'` +
+      buildGradle.slice(dependenciesStartIndex + 1);
+    await fs.writeFile(buildGradlePath, newBuildGradle);
+  }
+
+  public async listProviderPackages(): Promise<
+    { name: string; version: string }[]
+  > {
+    const dependencyRegex =
```

Comment:
There's also a [Gradle depedencies task](https://docs.gradle.org/current/userguide/viewing_debugging_dependencies.html) that might be more robust to use. But I don't have a strong opinion, so we might as well go with the regex until we hit problems.

Reply:
That one has no json output, so it felt as bad 😅